Sources/Swift/Integrations/Performance/IO/SentryFileIOTracker+SwiftHelpers.swift

```diff
@@ -0,0 +1,35 @@
+@_implementationOnly import _SentryPrivate
+
+extension SentryFileIOTracker {
+    func measureReadingData(from url: URL, options: Data.ReadingOptions, origin: String, method: (_ url: URL, _ options: Data.ReadingOptions) throws -> Data) throws -> Data {
+        guard url.scheme == NSURLFileScheme else {
+            return try method(url, options)
+        }
+        let span = self.startTrackingReadingFilePath(url.path, origin: origin)
+        defer {
+            self.endTrackingFile()
+        }
+        let result = try method(url, options)
+        if let span = span {
+            self.finishTrackingNSData(result, span: span)
+        }
+        return result
+    }
+
+    func measureWritingData(
+        _ data: Data,
+        to url: URL,
+        options: Data.WritingOptions,
+        origin: String,
+        method: (_ data: Data, _ url: URL, _ options: Data.WritingOptions) throws -> Void
+    ) throws {
+        guard url.scheme == NSURLFileScheme else {
+            return try method(data, url, options)
+        }
+        let span = self.startTrackingWriting(data, filePath: url.path, origin: origin)
+        try method(data, url, options)
+        if let span = span {
+            self.finishTrackingNSData(data, span: span)
```

Comment:
`h`: This must be in a defer, because otherwise if the method throws an error we never finish the span.

Reply:
In this case it is possible, because `data` is available.

Does the same apply to `measureReadingData`? The `finishTrackingNSData` requires a non-nil `Data`, which is not available if reading fails.

The Objective-C implementation calls [self finishTrackingNSData:data span:span]; in every case, even in an error where no `NSData`. It seems to me there is no test case to handle the case when an error is thrown.

Based on the [documentation](https://developer.apple.com/documentation/foundation/nsdata/1411145-initwithcontentsoffile) I assume it returns a zero bytes data, therefore I could use empty data as a nil fallback

Reply:
> Does the same apply to measureReadingData?

Yes, it should. You call try in that method, so it could exit before ending the span.

Reply:
Added a defer with `file.size` set to `0`. If you prefer to not set the `file.size` instead, let me know, I would leave it like that for now.